ha/core/controllers/pcs/pcs_controller.py

```diff
@@ -24,4 +32,66 @@ def __init__(self):
+        while True:
+            time.sleep(10)
+            _output, _err, _rc = self._execute.run_cmd(const.PCS_FAILCOUNT_STATUS,
+                                                       check_error=False)
+            if _output == const.NO_FAILCOUNT:
+                failcount_found = False
+            else:
+                failcount_found = True
+
+            if count >= const.RETRY_COUNT:
+                break
+            if failcount_found:
+                _output, _err, _rc = self._execute.run_cmd(const.PCS_CLEANUP,
+                                                               check_error=False)
+            count += 1
+        return failcount_found
+
+    def nodes_status(self, nodeids: list = None) -> dict:
+        """
+        Get pcs status of nodes.
+        Args:
+            nodeids (list): List of Node IDs from cluster nodes.
+                Default provide list of all node status.
+                if 'local' then provide local node status.
+
+        Returns:
+            ([dict]): Return dictionary. {"status": "", "msg":{}}}
+                status: Succeeded, Failed, InProgress
+        """
+        all_nodes_status = dict()
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_STATUS_NODES, check_error=False)
+        if nodeids is not None and isinstance(nodeids, list):
+            for nodeid in nodeids:
+                if nodeid in _output:
+                    for status in _output.split("\n"):
+                        nodes = status.split(":")
+                        if len(nodes) > 1 and nodeid.lower() in nodes[1].strip().lower():
+                            if nodes[0].strip().lower() == NODE_STATUSES.STANDBY.value.lower():
+                                all_nodes_status[nodeid] = NODE_STATUSES.STANDBY.value
+                            elif nodes[0].strip().lower() == NODE_STATUSES.STANDBY_WITH_RESOURCES_RUNNING.value.lower():
+                                all_nodes_status[nodeid] = NODE_STATUSES.STANDBY_WITH_RESOURCES_RUNNING.value
+                            elif nodes[0].strip().lower() == NODE_STATUSES.MAINTENANCE.value.lower():
+                                all_nodes_status[nodeid] = NODE_STATUSES.MAINTENANCE.value
+                            elif nodes[0].strip().lower() == NODE_STATUSES.OFFLINE.value.lower():
+                                all_nodes_status[nodeid] = NODE_STATUSES.OFFLINE.value
+                            elif nodes[0].strip().lower() == NODE_STATUSES.ONLINE.value.lower():
+                                all_nodes_status[nodeid] = NODE_STATUSES.ONLINE.value
+                            break
+                    else:
+                        all_nodes_status[nodeid] = NODE_STATUSES.UNKNOWN.value
+                else:
+                    raise HAInvalidNode(f"Node {nodeid} is not a part of cluster")
+            return all_nodes_status
+        else:
+            raise ClusterManagerError("Either nodeids is None or not a list to check the status")
```

Comment:
Remove default value of node_ids as None if it's not expected.
Also, you have added type hint for nodeids so why is this checking is required?

Reply:
If anyone pass the nodeids as string then the for loop executes for each character of string. Hence added that check.